.github/workflows/scripts/readme.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/zsh -ex
+
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+dirs=(`find . -type f -name "go.mod" -not -path "*/internal/*" -exec dirname {} \; | sort | egrep  '^./'`)
```

Comment:
This finds non-internal Go modules which is a subset of all non-internal Go packages. For instance https://pkg.go.dev/go.opentelemetry.io/otel/metric/noop is also a package.

Reply:
Hmm, I thought we'd want to do this for modules, not every package. That seems a tad too much?

Reply:
For a regular developer browsing GitHub there is not a big difference between a package and a module.

Reply:
I don't mind adding it, though it adds quite a lot of readme files.
But the main difference between both may be that it's more likely that folks who just get started with the library would want to use modules than each package?

Reply:
I've done it. One issue there is that now, we always have issues with the links checker, as we reach the `pkg.go.dev` requests limit.

Reply:
AFAIK it is possible to configure `lychee` to ignore `429` errors.

I also see that the specification repo uses a different action for checking hyperlinks and retries on `429`. See:
- https://github.com/open-telemetry/opentelemetry-specification/blob/65875014e15fa9739501413b3349961d503c0630/.github/workflows/checks.yaml#L36-L46
- https://github.com/open-telemetry/opentelemetry-specification/blob/65875014e15fa9739501413b3349961d503c0630/.markdown_link_check_config.json#L17

Reply:
Related issue https://github.com/lycheeverse/lychee/issues/989

Reply:
Going to a very low max concurrency of 5 seems to have fixed it for now.